--- a/src/VBox/NetworkServices/NAT/pxping.c
+++ b/src/VBox/NetworkServices/NAT/pxping.c
@@ -63,7 +63,7 @@
 #include "lwip/icmp.h"
 
 #if defined(RT_OS_LINUX) && !defined(__USE_GNU)
-#if __GLIBC_PREREQ(2, 8)
+#if __MUSL_PREREQ
 /*
  * XXX: This is gross.  in6_pktinfo is now hidden behind _GNU_SOURCE
  * https://sourceware.org/bugzilla/show_bug.cgi?id=6775
--- a/src/VBox/Runtime/r3/linux/semevent-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/semevent-linux.cpp
@@ -35,7 +35,7 @@
  */
 
 #include <features.h>
-#if __GLIBC_PREREQ(2,6) && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
+#if __MUSL_PREREQ && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
 
 /*
  * glibc 2.6 fixed a serious bug in the mutex implementation. We wrote this
--- a/src/VBox/Runtime/r3/linux/semeventmulti-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/semeventmulti-linux.cpp
@@ -36,7 +36,7 @@
 
 
 #include <features.h>
-#if __GLIBC_PREREQ(2,6) && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
+#if __MUSL_PREREQ && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
 
 /*
  * glibc 2.6 fixed a serious bug in the mutex implementation. We wrote this
--- a/src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp
@@ -42,7 +42,7 @@
 # define _GNU_SOURCE
 #endif
 #include <features.h>
-#if __GLIBC_PREREQ(2,4)
+#if __MUSL_PREREQ
 
 #include <sched.h>
 #include <unistd.h>